include/devicetree/fixed-partitions.h

```diff
@@ -79,6 +79,23 @@ extern "C" {
 		    (DT_PARENT(DT_GPARENT(node_id))),			\
 		    (DT_GPARENT(node_id)))
 
+/*
+ * @brief Get the common mount flags for an fstab entry.
+
+ * @param node_id the node identifier for a child entry in a
+ * zephyr,fstab node.
+ * @return a value suitable for initializing an fs_mount_t flags
+ * member.
+ */
+#define DT_FSTAB_ENTRY_MOUNT_FLAGS(node_id) 		\
```

Comment:
This is a layering violation; it should be in include/fs/fs.h. The FS_MOUNT_FLAG_FOO values are not available here.

Reply:
No, but they're not depended on here either.  `<fs/fs.h>` doesn't even needed to be included first for this to work.

This seems to be a chicken/egg problem.  I think this macro needs to be `DT_something` since it's converting devicetree node data to what the code requires.  Do you prefer that it be defined in `<fs/fs.h>` rather than `<devicetree/something.h>`, or should it not have a `DT_` prefix?

Reply:
I think putting it in `fs.h` as `FS_DT_MOUNT_FLAGS(node_id)` or so should solve the problem. It shouldn't start with `DT_` since it's not in `devicetree.h` or a private subheadher, but it should have `DT` somewhere in the name.